File: src/sections/weather.js
const https = require('https');

function fetch(url) {
  return new Promise((resolve, reject) => {
    https.get(url, { headers: { 'User-Agent': 'daily-briefing/1.0' } }, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        if (res.statusCode >= 400) {
          reject(new Error(`HTTP ${res.statusCode}: ${data.slice(0, 200)}`));
        } else {
          resolve(data);
        }
      });
    }).on('error', reject);
  });
}

async function getWeather(args) {
  const location = encodeURIComponent(args.location);
  const url = `https://wttr.in/${location}?format=j1`;
  const raw = await fetch(url);
  const data = JSON.parse(raw);

  const current = data.current_condition?.[0] || {};
  const today = data.weather?.[0] || {};
  const hourly = today.hourly || [];

  // Find afternoon temp (around 3pm)
  const afternoon = hourly.find(h => h.time === '1500') || hourly[hourly.length - 1] || {};

  return {
    location: args.location,
    condition: current.weatherDesc?.[0]?.value || 'Unknown',
    temp_f: current.temp_F || '?',
    temp_c: current.temp_C || '?',
    feels_like_f: current.FeelsLikeF || '?',
    humidity: current.humidity || '?',
    wind_mph: current.windspeedMiles || '?',
    wind_dir: current.winddir16Point || '',
    high_f: today.maxtempF || '?',
    low_f: today.mintempF || '?',
    high_c: today.maxtempC || '?',
    low_c: today.mintempC || '?',
    uv_index: current.uvIndex || '?',
    sunrise: today.astronomy?.[0]?.sunrise || '?',
    sunset: today.astronomy?.[0]?.sunset || '?',
    precip_chance: afternoon.chanceofrain || '0',
  };
}

module.exports = { getWeather };

File: daily_briefing/sections/weather.py
"""Weather section — fetches current conditions from wttr.in."""

import json
import urllib.request
import urllib.parse


def get_weather(args):
    location = urllib.parse.quote(args.location)
    url = f"https://wttr.in/{location}?format=j1"
    req = urllib.request.Request(url, headers={"User-Agent": "daily-briefing/2.0"})

    with urllib.request.urlopen(req, timeout=15) as resp:
        data = json.loads(resp.read().decode())

    current = data.get("current_condition", [{}])[0]
    today = data.get("weather", [{}])[0]
    hourly = today.get("hourly", [])

    # Find afternoon temp (~3pm)
    afternoon = next((h for h in hourly if h.get("time") == "1500"), hourly[-1] if hourly else {})

    return {
        "location": args.location,
        "condition": (current.get("weatherDesc", [{}])[0]).get("value", "Unknown"),
        "temp_f": current.get("temp_F", "?"),
        "temp_c": current.get("temp_C", "?"),
        "feels_like_f": current.get("FeelsLikeF", "?"),
        "humidity": current.get("humidity", "?"),
        "wind_mph": current.get("windspeedMiles", "?"),
        "wind_dir": current.get("winddir16Point", ""),
        "high_f": today.get("maxtempF", "?"),
        "low_f": today.get("mintempF", "?"),
        "high_c": today.get("maxtempC", "?"),
        "low_c": today.get("mintempC", "?"),
        "uv_index": current.get("uvIndex", "?"),
        "sunrise": (today.get("astronomy", [{}])[0]).get("sunrise", "?"),
        "sunset": (today.get("astronomy", [{}])[0]).get("sunset", "?"),
        "precip_chance": afternoon.get("chanceofrain", "0"),
    }
